File: check_item.go
// Random Go struct generated on 2025-06-27
package main

import (
    "encoding/json"
    "fmt"
    "time"
)

type UpdateUser struct {
    Name      string    `json:"name"`
    Value     int       `json:"value"`
    Active    bool      `json:"active"`
    CreatedAt time.Time `json:"created_at"`
}

func NewUpdateUser(name string, value int) *UpdateUser {
    return &UpdateUser{
        Name:      name,
        Value:     value,
        Active:    true,
        CreatedAt: time.Now(),
    }
}

func (s *UpdateUser) UpdateValue(newValue int) {
    s.Value = newValue
}

func (s *UpdateUser) ToJSON() ([]byte, error) {
    return json.Marshal(s)
}

func main() {
    obj := NewUpdateUser("sample", 65)
    data, _ := obj.ToJSON()
    fmt.Println(string(data))
}

File: check_item.ts
// Random TypeScript class generated on 2025-11-21

class SetValue {
    private value: number;
    private readonly createdAt: Date;

    constructor(initialValue: number = 49) {
        this.value = initialValue;
        this.createdAt = new Date();
    }

    public getValue(): number {
        return this.value;
    }

    public updateValue(newValue: number): void {
        this.value = newValue;
    }

    public getInfo(): object {
        return {
            value: this.value,
            createdAt: this.createdAt.toISOString()
        };
    }
}

export { SetValue };
